cf-execd/cf-execd.c

```diff
@@ -434,6 +434,17 @@ void ThisAgentInit(void)
     umask(077);
 }
 
+// Return 2 to 62 seconds depending on how much time is left until the next
+// minute starts. Since cf-execd wakes up "every minute" to evaluate its
+// schedule, we want to do this at the start of the minute, to avoid
+// accidentally skipping any runs if things are slow.
+static inline time_t GetPulseTime()
+{
+    const time_t current_second = (time(NULL) % 60);
```

Comment:
@larsewi [mentioned](https://github.com/cfengine/core/pull/5499/files#r1592600540) using `CFPULSETIME` here instead of 60 but if we want to start at the beginning of every minute I would say remove `CFPULSETIME` entirely and keep these hard-coded 60s here.

Reply:
Okay, adjusted to use the constant.

Reply:
This change actually doesn't make sense to me. If we want to determine which second in a minute the clock currently is, `% 60` is the right calculation. If `CFPULSETIME` is set to 120 or 180, its use in the calculation won't make sense.

Reply:
I thought a bit about that, but it does make some sense. If you define a 120 second pulse, it will tell you how many seconds is left of the current pulse, even though that feels less intuitive to think about than the current setup.